test/topology/conftest.py

```diff
@@ -168,18 +180,44 @@ async def manager_internal(event_loop, request):
     await manager_int.stop()  # Stop client session and close driver after last test
 
 @pytest.fixture(scope="function")
-async def manager(request, manager_internal):
+async def manager(request, manager_internal, record_property, mode):
     """Per test fixture to notify Manager client object when tests begin so it can
     perform checks for cluster state.
     """
     test_case_name = request.node.name
-    await manager_internal.before_test(test_case_name)
+    run_id = request.config.getoption('run_id')
+    tmp_dir = pathlib.Path(request.config.getoption('tmpdir'))
+    xml_path = pathlib.Path(request.config.getoption('xmlpath'))
+    suite_testpy_log = tmp_dir / mode / f"{xml_path.name[:-10]}.log"
+    test_log = tmp_dir / mode / f"{suite_testpy_log.name[:-4]}.{test_case_name}.log"
+
+    await manager_internal.before_test(test_case_name, test_log)
     yield manager_internal
     # `request.node.stash` contains a flag stored in `pytest_runtest_makereport`
     # that indicates test failure.
     failed = request.node.stash[FAILED_KEY]
+    if failed:
+        # Save scylladb logs for failed tests in a separate directory and copy XML report to the same directory to have
+        # all related logs in one dir.
+        # Then add property to the XML report with the path to the directory, so it can be visible in Jenkins
+        failed_test_dir_path = tmp_dir / mode / "failed_test" / f"{test_case_name}.{mode}.{run_id}"
+        failed_test_dir_path.mkdir(parents=True, exist_ok=True)
+        await manager_internal.gather_related_logs(
+            failed_test_dir_path,
+            {'pytest.log':test_log, 'test_py.log': suite_testpy_log}
+        )
+        with open(failed_test_dir_path / f"stacktrace", 'w') as f:
+            f.write(request.node.stash[phase_report_key].longreprtext)
+        if request.config.getoption('artifacts_dir_url'):
+            # get the relative path to the tmpdir for the failed directory
+            dir_path_relative = f"{failed_test_dir_path.as_posix()[failed_test_dir_path.as_posix().find('testlog'):]}"
```

Comment:
is there no better way to calculate this (using some nice high-level methods from `Path`) instead of these string manipulations?

Reply:
The current way of calculating the path is fragile and clumsy, but I could not find a better solution. An alternative solution was introducing additional parameter to the test.py and providing it during the run, but I'm not a big fan of such an idea.

Reply:
ok, so let's leave it like that